### DISCUSSION TITLE
Obtaining the convex hull of large number of convex sets

### DISCUSSION THREAD

**Discussion by gayanlanke** [yalmip/YALMIP#1088] [General discussion] [answered]
I have 5 constrained sets,  actually convex sets, and I need to obtain the convex hull of them. So I follow an approach as below.

```
clc;clear

sdpvar x y

F1 = [-1 <= x <= 1, -1 <= y <= 1];  
F2 = [-3 <= x <= -2, -2 <= y <= 2];
F3 = [-5 <= x <= -4, -3 <= y <= 3];
F4 = [-9 <= x <= -6, -5 <= y <= 5];

H=hull(F1,F2,F3,F4);  %% convex hull
```

This is perfectly fine, and I can get the convex hull without any problem. Imagine I have 100 constrained sets for example, do I need to still use the command `hull(F1,F2,F3,....F100)` (tedious of course!) to obtain the convex hull?

Instead of typing `F1,F2,...F100`, is it possible to define a vector or a cell array `F_vec` and get the convex hull as follows:

```
H=hull(F_vec)
```
Thanks in advance.

**Comment by johanlofberg** [comment, marked as answer]
You are generating them somewhere, so simply put all in a cell and then do hull(these{:})

Although it sounds like a very expensive representation if the sets actually look like this. Just compute all the vertices explicitly and then corresponding convex hull vertices, and then generate the constraint

**Reply by gayanlanke** [reply]
Thanks a lot, it worked.

For *"Although it sounds like a very expensive representation if the sets actually look like this. Just compute all the vertices explicitly and then corresponding convex hull vertices, and then generate the constraint"*, although I represent them in this form, I already have the convex hull and then obtain `A` and `b` to represent each as hyperspaces given by `A*x<=b`. I guess that should be okay.

**Reply by johanlofberg** [reply]
Not sure what you are saying, but the constraints you show here have 4 vertices, and computing the convex hull of them is a simple call to convhull and then use a convex hull representation of those vertices, or if you want the facet-representation of the convex hull, it is also easily computed by said command as it is in 2D and thus tractable
